satellite/audit/pathcollector.go

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package audit
+
+import (
+	"context"
+	"math/rand"
+
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
+)
+
+// PathCollector uses the metainfo loop to add paths to node reservoirs
+type PathCollector struct {
+	Reservoirs     map[storj.NodeID]*Reservoir
+	reservoirSlots int
+	rand           *rand.Rand
+}
+
+// NewPathCollector instantiates a path collector
+func NewPathCollector(reservoirSlots int, r *rand.Rand) *PathCollector {
+	return &PathCollector{
+		Reservoirs:     make(map[storj.NodeID]*Reservoir),
+		reservoirSlots: reservoirSlots,
+		rand:           r,
+	}
+}
+
+// RemoteSegment takes a remote segment found in metainfo and creates a reservoir for it if it doesn't exist already
+func (pathCollector *PathCollector) RemoteSegment(ctx context.Context, path storj.Path, pointer *pb.Pointer) (err error) {
```

Comment:
you can name the receiver `collector` I think it's sufficiently descriptive in this context